src/main.py

```diff
@@ -0,0 +1,36 @@
+# pylint: disable=import-outside-toplevel
+import asyncio
+import sys
+from pathlib import Path
+
+
+def main(script_root: Path):
+    try:
+        from src.cli import ArgumentDefaultValueFromConfigProvider, ArgumentParserFacade
```

Comment:
is top level import causing a import loop?

Reply:
Why it does?

Reply:
Idk. GitPython throws an error during importing. I've just noticed this error isn't caught here, so I'll move try/except to the `protostar.py`.

However, I'll leave importing `ArgumentParserFacade` here because import patching in `main_test.py` stops working when top-level import is used and I didn't figure it out why it behaves like that.

Reply:
Isn't the same issue with name shadowing we had once already? I think we have to figure it out instead of making dirty workaround. If you want my help with it let me know

Reply:
Yep, I need help with it. It looks like a different issue. I spent already too much time on it and ran out of ideas. I think we should try to fix it in a different PR.

Reply:
Ok, please finish the requested changes and I will take a look on it